packages/core/src/actions.ts

```diff
@@ -159,6 +159,12 @@ export function toActivityDefinition<TContext, TEvent extends EventObject>(
  *
  * @param eventType The event to raise.
  */
+export function raise<TEvent extends EventObject>(
```

Comment:
could you add some tests for this? it's hard to review this as it's not obvious what does this fix exactly

Reply:
Might be tricky; but I noticed it in places like this:

<img width="580" alt="CleanShot 2021-07-01 at 09 29 17@2x" src="https://user-images.githubusercontent.com/1093738/124132663-0d93e580-da4f-11eb-95b6-cff9607215cd.png">

Reply:
@Andarist These fix `packages/core/test/types.test.ts` (see below)

Reply:
I wouldn't exactly say that those mentioned call sites were fixed. This, somehow, made just any event accepted at those positions.

<img width="721" alt="Screenshot 2022-12-03 at 17 55 53" src="https://user-images.githubusercontent.com/9800850/205452387-8d95a7e9-2bf0-4db9-a300-31970759c01c.png">

This is most likely related to the fact that our types are incorrect here:
https://github.com/statelyai/xstate/blob/b9995f0844bbbff0c813fee020935dfd7562184b/packages/core/src/types.ts#L227

The problem is that we should have 2 event-related generics flying around here.

One should be the filtered down `TEventExpression` that would come from this `Extract` here:
https://github.com/statelyai/xstate/blob/b9995f0844bbbff0c813fee020935dfd7562184b/packages/core/src/types.ts#L458
and the other one should just be `TEvent` coming from the top-level.

One is about what the given expression can receive and the other one is about what raise can "dispatch"